main.go

```diff
@@ -85,12 +85,13 @@ func getBlockTime(height int64) (int64, error) {
 	return block.Time, nil
 }
 
-func binarySearch(blockCount int64, targetTime int64) int64 {
+func binarySearch(blockCount int64, targetTime int64) string {
 	// Check if the result is already cached
 	cacheMutex.RLock()
 	if cachedEntry, ok := cacheMap[targetTime]; ok {
 		cacheMutex.RUnlock()
-		return cachedEntry.height
+		resultStr := strconv.FormatInt(cachedEntry.height, 10)
```

Comment:
Stylistic recommendation:  the variable name can be `result` rather than `resultStr`.  As a rule of thumb, you don't need to specify the type in the variable name, especially in a language like Go which has strong static typing.